### ISSUE TITLE
[FR] (add an auto home before the leveling )

### ISSUE THREAD

**Issue by Nicog77** [smith3d/Marlin#19]
Actually when using the level button, the printing head move from the center to the front left border of the bed.
If you stopper steppers and moving manually the nozzle, and then pushing the level button, the head will be moving at bad coordonates or the motor arrives to the X or Y limits and make some weird/horrible noises and lose steps.

Adding an autohome before will do the level function safer for the printer.

### Description

Autohome to the X and Y axis before starting the Z auto probing with the bltouch

### Feature Workflow

1. when user push the level button, the printer check its coordonates by doing an autohome on X and Y axis
2. after testing the X and Y axis the printer starts its Z axis probing

#### Additional Information

Thanks

**Comment by tfaz099** [comment]
@Nicog77 Yes this is a great feature request.  I too have done a physical bed leveling and then forgot to AutoHome and ended up with the Automatic Leveling start in the middle of the bed and try to run off the x-axis LOL.  Adding AutoHome to the start of Automatic Leveling would be a definite plus.
